src/init.cpp

```diff
@@ -1526,6 +1528,46 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
             // Create new keyUser and set as default key
             RandAddSeedPerfmon();
 
+            if (GetBoolArg("-usehd", DEFAULT_USE_HD_WALLET))
```

Comment:
I'd still prefer the user is warned that they are trying to use hd but already have a running non-hd wallet or simply be disallowed.